### PR TITLE
fix 10094

### PR DESCRIPTION
**Pull request by saeideng**
use default options within __invoke()
Refs: #10094

### CODE REVIEW

**Comment by markstory** [comment]
How does this fix the issue? Could we get at test?

File: src/ORM/Rule/IsUnique.php [hunk 1]
```diff
@@ -67,19 +67,20 @@ public function __invoke(EntityInterface $entity, array $options)
         if (!$entity->extract($this->_fields, true)) {
             return true;
         }
+        $this->_options = $options + $this->_options;
```
Comment (markstory) [inline src/ORM/Rule/IsUnique.php]: I'm not sure we should be mutating state on each call. Could this be `$options = $options + $this->_options`?

**Review by markstory — CHANGES_REQUESTED** [review]

**Comment by saeideng** [comment]
option sent in 
https://github.com/cakephp/cakephp/blob/master/src/ORM/Table.php#L2331

> How does this fix the issue? Could we get at test?

i,m not time now , but maybe 1,2 days later ,have someone that can will add test?

**Comment by markstory** [comment]
> have someone that can will add test?

I would be happy to add a test, bu I don't even know what is going wrong here.

**Comment by saeideng** [comment]
see https://github.com/cakephp/cakephp/pull/9704
when _invoke is run `$options['repository']` is empty
Because you not sent `repository` key see https://github.com/cakephp/cakephp/blob/master/src/ORM/Table.php#L2333

**Comment by saeideng** [comment]
you are right, this PR could not fixed that problem :(

**Pull request closed without merging**